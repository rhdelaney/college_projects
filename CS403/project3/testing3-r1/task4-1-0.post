
(define myheight (make-connector))
(define mylat (make-connector))
(define myspeed (make-connector))
(define (get-value-safe c) (if (has-value? c) (get-value c) 'no-value))

(speed myspeed mylat myheight)

(println "setting latitude")
(set-value! mylat 61 'user)
(inspect (get-value-safe mylat))
(inspect (get-value-safe myspeed))
(inspect (get-value-safe myheight))
(println "setting velocity")
(set-value! myspeed 11997 'user)
(inspect (get-value-safe mylat))
(inspect (get-value-safe myspeed))
(inspect (get-value-safe myheight))
(println "forgetting speed")
(forget-value! myspeed 'user)
(inspect (get-value-safe mylat))
(inspect (get-value-safe myspeed))
(inspect (get-value-safe myheight))
(println "setting height")
(set-value! myheight .12 'user)
(inspect (get-value-safe mylat))
(inspect (get-value-safe myspeed))
(inspect (get-value-safe myheight))
(println "forgetting latitude")
(forget-value! mylat 'user)
(inspect (get-value-safe mylat))
(inspect (get-value-safe myspeed))
(inspect (get-value-safe myheight))
(println "setting latitude")
(set-value! mylat 4 'user)
(inspect (get-value-safe mylat))
(inspect (get-value-safe myspeed))
(inspect (get-value-safe myheight))
