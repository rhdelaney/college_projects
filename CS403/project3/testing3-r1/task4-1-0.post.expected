setting latitude
(get-value-safe mylat) is 61
(get-value-safe myspeed) is no-value
(get-value-safe myheight) is no-value
setting velocity
(get-value-safe mylat) is 61
(get-value-safe myspeed) is 11997
(get-value-safe myheight) is 0.6650172861
forgetting speed
(get-value-safe mylat) is 61
(get-value-safe myspeed) is no-value
(get-value-safe myheight) is no-value
setting height
(get-value-safe mylat) is 61
(get-value-safe myspeed) is 5096.2041215
(get-value-safe myheight) is 0.1200000000
forgetting latitude
(get-value-safe mylat) is no-value
(get-value-safe myspeed) is no-value
(get-value-safe myheight) is 0.1200000000
setting latitude
(get-value-safe mylat) is 4
(get-value-safe myspeed) is 5085.9376009
(get-value-safe myheight) is 0.1200000000
